tests/web/Widget.cc

```diff
@@ -0,0 +1,146 @@
+//  This file is part of Empirical, https://github.com/devosoft/Empirical
+//  Copyright (C) Michigan State University, 2020.
+//  Released under the MIT Software license; see doc/LICENSE
+
+#include <functional>
+#include <unordered_map>
+
+#include "base/assert.h"
+#include "web/testing/TestManager.h"
+#include "web/Document.h"
+#include "web/Element.h"
+#include "web/web.h"
+
+
+struct Test_WidgetWrapWith : emp::web::BaseTest {
+  emp::web::Document doc{"emp_test_container"};
+
+  void Setup() override {
+    // Construct the following HTML structure using empirical
+    // <div id="wrapper2"> <!-- inserted with WrapWith -->
+    //   <p id="parent">
+    //     parent
+    //     <div id="wrapper"> <!-- inserted with WrapWith -->
+    //       wrapper
+    //       <button id="child"></button>
+    //     </div>
+    //   </p>
+    //   <br/><br/>
+    // </div>
+
+    // Manually activate document (and re-trigger ready signal)
+    doc.Activate();
```

Comment:
We could add this as a macro/utility function (and I think it'd be super useful). 

I'm not sure it belongs in the MochaTestRunner. I'd lean toward a function that lives in Document.h or init.h maybe. 

I'm going to say this falls outside the scope of this pull request, and I'll file an issue/feature request for this. It doesn't seem like a difficult add, but I think we want to talk about how this should look/where it should live before adding it.

Reply:
not just doc.Activate() but also EM_ASM({jquery.ready()}

actually, as a better idea could we just put these two snippets into the constructor for BaseTest??? That way it would be magically run and the end user would never have to think about or maintain it

Reply:
putting it inside BaseTest means that we'd have to add a Document member variable to BaseTest, which I don't super want to do (I don't think we should assume every test will want/need a single document object hanging out in the base class). 

I like either adding a utility function that runs those snippets (probably should live in Document.h), or we should take a look at Document and see if we can make it a bit more clever with respect to how it gets initialized.